lib/api/core/sdk/funnelProtocol.js

```diff
@@ -22,23 +22,46 @@
 'use strict';
 
 const
-  KuzzleEventEmitter = require('kuzzle-sdk').KuzzleEventEmitter,
-  Request = require('kuzzle-common-objects').Request;
+  {
+    Request,
+    errors: {
+      InternalError: KuzzleInternalError
+    }
+  } = require('kuzzle-common-objects'),
+  User = require('../models/security/user'),
+  KuzzleEventEmitter = require('kuzzle-sdk').KuzzleEventEmitter;
 
 class FunnelProtocol extends KuzzleEventEmitter {
-  constructor(funnel) {
+  constructor(funnel, user = null) {
     super();
+
     this.funnel = funnel;
     this.id = 'funnel';
+    this.user = user;
+
+    if (this.user && !(this.user instanceof User)) {
+      throw new KuzzleInternalError('FunnelProtocol.constructor: Invalid User object');
```

Comment:
(nitpicking) I agree with that check, but if we do get that error in production, then it's probably a good idea to add the received `user` argument in the error message, to make debugging easier.